pandas/core/arrays/datetimelike.py

```diff
@@ -1435,6 +1435,50 @@ def max(self, axis=None, skipna=True, *args, **kwargs):
         # Don't have to worry about NA `result`, since no NA went in.
         return self._box_func(result)
 
+    def mean(self, axis=None, skipna=True):
+        """
+        Return the mean value of the Array or mean along an axis.
+
+        .. versionadded:: 0.25.0
+
+        Parameters
+        ----------
+        axis : None
+            Dummy parameter to match NumPy signature
+        skipna : bool, default True
+            Whether to ignore any NaT elements
+
+        See Also
+        --------
+        numpy.ndarray.mean
+        Series.mean : Return the mean value in a Series.
+        """
+        if is_period_dtype(self):
+            # See discussion in GH#24757
+            raise NotImplementedError(
```

Comment:
NotImplementedError seems to indicate that we just not *yet* implemented it, while here we actually explicitly don't want to calculate it. Would a TypeError be more suitable?